lib/Tests/Fhp/Protocol/DialogInitializationTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Tests\Fhp\Protocol;
+
+use Fhp\Protocol\DialogInitialization;
+
+class DialogInitializationTest extends \PHPUnit\Framework\TestCase
+{
+    public function testSerializableInterfaceMigration()
+    {
+        $kundensystemId = 'kunden-system-id';
+        $needTanForSegment = 'test-segment';
+
+        $object = new DialogInitializationTestModel(
+            $kundensystemId,
+            $needTanForSegment,
+        );
+
+        $string = serialize($object);
+
+        /** @var DialogInitialization $object2 */
+        $object2 = unserialize($string);
+        self::assertTrue(is_object($object2));
```

Comment:
Use assertIsObject, assertNotSame, assertEquals here and below. If the test fails, phpUnit will print the actual values which helps debugging (especially on Travis, where one can't just run a step-by-step debugger to inspect variable values).